src/Controls/src/Core/Registrar.cs

```diff
@@ -79,6 +82,37 @@ public void Register(Type tview, Type trender, Type[] supportedVisuals, short pr
 		/// <include file="../../docs/Microsoft.Maui.Controls.Internals/Registrar.xml" path="//Member[@MemberName='Register']/Docs" />
 		public void Register(Type tview, Type trender) => Register(tview, trender, _defaultVisualRenderers);
 
+		HashSet<Type> _registeredVisuals = new HashSet<Type>();
+
+		internal void RegisterVisual([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] Type registeredVisual) => _registeredVisuals.Add(registeredVisual);
+
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
+		internal Type GetVisual(string visualName)
+		{
+			foreach (var visualType in _registeredVisuals)
+			{
+				string name = visualType.Name;
+				string fullName = visualType.FullName;
+
+				if (name.EndsWith("Visual", StringComparison.OrdinalIgnoreCase))
+				{
+					name = name.Substring(0, name.Length - 6);
+					fullName = fullName.Substring(0, fullName.Length - 6);
```

Comment:
Can we use `ReadOnlySpan<char>` and Slice here instead? Then below, when doing the comparisons, use `SequenceEqual` instead of `==`.

And also not creating a new string appending `Visual` below. Just slice off "Visual" from `visualName`.